compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -693,7 +694,7 @@ abstract class PrepJSInterop extends plugins.PluginComponent
         }
 
         val ownerLoadSpec = jsInterop.jsNativeLoadSpecOf(sym.owner)
-        val jsName = jsInterop.jsNameOf(sym)
+        val JSName.Literal(jsName) = jsInterop.jsNameOf(sym)
```

Comment:
In the current state, I expect this line to throw a `MatchError` for the following:

```scala
@js.native
object Foo extends js.Any {
  @JSName(sym)
  @js.native
  class A extends js.Any
}
```